docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
 
-### NotificationsClient
+### DeviceServiceClient()
+Returns a `DeviceServiceClient` to leverage Core Metadata's `DeviceService` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
 
-The `NotificationsClient` exposed on the context is available to leverage Support Notifications' `Notifications` API. See [README](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/notifications/README.md) for more details. Useful for sending notifications. 
+### DeviceProfileClient()
+Returns a `DeviceProfileClient` to leverage Core Metadata's `DeviceProfile` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
+
+### DeviceClient()
+Returns a `DeviceClient` to leverage Core Metadata's `Device` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
 
 ### Note about Clients
 
 Each of the clients above is only initialized if the Clients section of the configuration contains an entry for the service associated with the Client API. If it isn't in the configuration the client will be `nil`. Your code must check for `nil` to avoid panic in case it is missing from the configuration. Only add the clients to your configuration that your Application Service will actually be using. All application services need `Core-Data` for version compatibility check done on start-up. The following is an example `Clients` section of a configuration.toml with all supported clients specified:
 
 ```
 [Clients]
-  [Clients.Logging]
-  Protocol = "http"
-  Host = "localhost"
-  Port = 48061
 
-  [Clients.CoreData]
+  [Clients.core-data]
+  Protocol = 'http'
+  Host = 'localhost'
+  Port = 59880
+  
+  [Clients.core-metadata]
   Protocol = 'http'
   Host = 'localhost'
-  Port = 48080
+  Port = 59881
 
-  [Clients.Command]
+  [Clients.core-command]
   Protocol = 'http'
   Host = 'localhost'
-  Port = 48082
+  Port = 59882
 
-  [Clients.Notifications]
+  [Clients.support-notifications]
   Protocol = 'http'
   Host = 'localhost'
-  Port = 48060
+  Port = 59860
 ```
 
 ## .PushToCore()
-`.PushToCore(string deviceName, string readingName, byte[] value)` is used to push data to EdgeX Core Data so that it can be shared with other applications that are subscribed to the message bus that core-data publishes to. `deviceName` can be set as you like along with the `readingName` which will be set on the EdgeX event sent to CoreData. This function will return the new EdgeX Event with the ID populated, however the CorrelationId will not be available.
+`.PushToCore(event dtos.Event)` is used to push data to EdgeX Core Data so that it can be shared with other applications that are subscribed to the message bus that core-data publishes to. This function will return the new EdgeX Event with the ID populated, along with any error encountered.  Note that CorrelationId will not be available.
```

Comment:
```suggestion
`PushToCore(event dtos.Event)` is used to push data to EdgeX Core Data so that it can be shared with other applications that are subscribed to the message bus that core-data publishes to. This function will return the new EdgeX Event with the ID populated, along with any error encountered.  Note that CorrelationId will not be available.
```